src/DataFrames.jl

```diff
@@ -128,4 +128,18 @@ for (dir, filename) in [
     include(joinpath(dir, filename))
 end
 
+# Import  "StringEncodings" if available and set up decoder
+_decoder=nothing
+ENCODING_PACKAGE_READY = false
+function __init__()
+    try Pkg.installed("StringEncodings")
```

Comment:
> I might be missing something here as to the meaning of `LOAD_PATH`.

You probably don't have any packages in `LOAD_PATH`, it doesn't mean others don't.

> Just to make sure, you are saying the packages the require DataFrames will broken. right?

Yes.

> Are you saying a precompiled package with dependencies will not be calling the init of its dependent packages when loading.

It will, and that's the problem, since that's a unpredictable action during precompilation.

> Or, more concretely, if a package X depends on DataFrames that was initially installed without StringEncodings. After installing StringEncodings, package X will still not be able to use feature in DataFrames that StringEncodings enable. Right?

The precise symptom will not be very simple. It can cause this for sure in some cases.